### PR TITLE
shell: Add .connect meta command

### PR DESCRIPTION
**Pull request by theopolis**
This adds a helpful debugging feature to the shell, the ability to run queries against other osquery processes.

### CODE REVIEW

File: docs/wiki/deployment/debugging.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Here are some debugging tips and tricks related to the daemon and shell from a deployment and usage perspective. Please see the development documentation in the next section for debugging code changes.
 
-Almost every situation requiring debugging should ultimately be solved with bug fixes or better documentation. In these cases,  the documentation usually surfaces in the form of verbose messages in the tools.
+Almost every situation requiring debugging should ultimately be solved with bug fixes or better documentation. In these cases, the documentation usually surfaces in the form of verbose messages in the tools.
```
[the remark this answers is not in the document]
Comment (theopolis) [reply docs/wiki/deployment/debugging.md:5]: Are you OK with the current implementation supporting both `.connect` and `--connect`? Are you advocating that I remove one or the other?

[the remark this answers is not in the document]
Comment (directionless) [reply docs/wiki/deployment/debugging.md:5]: Ha. I missed the command line flag in this. I am agnostic about whether providing both will prove handy, or confusing. Seems fine to try.

[the remark this answers is not in the document]
Comment (directionless) [reply docs/wiki/deployment/debugging.md:5]: I also have no idea why I hung this comment here

**Review by directionless — APPROVED** [review]
Seems fine to me. Not sure if you want any of the deeper c++ people to look.

I built it, and tried various things. Seems to work as advertised. Amusingly, I can connect to the socket `osqueryi` itself created.

One caveat, is that the tab completion is for the locally loaded tables, and not the remote

**Pull request merged**